client/components/stats-interval-dropdown/index.jsx

```diff
@@ -77,7 +78,9 @@ const IntervalDropdown = ( { slug, period, queryParams, intervals, onGatedHandle
 	}
 
 	function getDisplayLabel() {
-		return intervals[ period ].label;
+		// If the period is not in the intervals list, capitalize it and show in the label - however user wouldn't be able to select it.
+		// TODO: this is a temporary solution, we should remove this once we have all the intervals in the list.
+		return intervals[ period ]?.label ?? capitalize( period );
```

Comment:
We intentionally didn't add Hour as a dropdown option, because we don't want to expose it, even after the date filtering project is released. Thoughts? 🤔